File: LCX/ValidEndPoints/dsl/examples/market_data_dynamic.nim
import httpclient
import json
import hmac
import base64


proc fetch_all_pairs_data*(pair): JsonNode =
  ## GET /api/ticker
  let url = "https://exchange-api.lcx.com/api/ticker"

  let client = newHttpClient()
  let response = client.get(url)
  let data = parseJson(response.body)
  echo data

  return data

# Workflow: FetchAllPairsData (sequence)

proc get_ethusdc_ticker*(pair): JsonNode =
  ## GET /api/ticker
  let url = "https://exchange-api.lcx.com/api/ticker"

  let client = newHttpClient()
  let response = client.get(url)
  let data = parseJson(response.body)
  echo data

  return data

proc get_lcxusdc_ticker*(pair): JsonNode =
  ## GET /api/ticker
  let url = "https://exchange-api.lcx.com/api/ticker"

  let client = newHttpClient()
  let response = client.get(url)
  let data = parseJson(response.body)
  echo data

  return data

File: LCX/ValidEndPoints/dsl/examples/market_data_dynamic.cpp
#include <iostream>
#include <curl/curl.h>
#include <nlohmann/json.hpp>

std::string fetch_all_pairs_data(pair) {
    std::string url = "https://exchange-api.lcx.com/api/ticker";
    // GET request implementation
    auto response = http_client.get(url);
    return response.body();
}
# Workflow: FetchAllPairsData (sequence)

std::string get_ethusdc_ticker(pair) {
    std::string url = "https://exchange-api.lcx.com/api/ticker";
    // GET request implementation
    auto response = http_client.get(url);
    return response.body();
}
std::string get_lcxusdc_ticker(pair) {
    std::string url = "https://exchange-api.lcx.com/api/ticker";
    // GET request implementation
    auto response = http_client.get(url);
    return response.body();
}
